src/sagemaker/model.py

```diff
@@ -728,13 +728,22 @@ def _compilation_job_config(
             "Framework": framework.upper(),
         }
 
-        multiple_version_supported_framework_list = ["pytorch", "tensorflow"]
-        if (
-            framework.lower() in multiple_version_supported_framework_list
-            and target_instance_type is not None
-            and re.match("(?=^ml_)(?!ml_inf)", target_instance_type) is not None
-            and framework_version is not None
+        def multi_version_compilation_supported(
+            target_instance_type: str, framework: str, framework_version: str
         ):
+            if target_instance_type and framework and framework_version:
+                framework = framework.lower()
+                multi_version_frameworks_support_mapping = {
+                    "ml_inf": ["pytorch", "tensorflow", "mxnet"],
+                    "ml_ioc": ["pytorch", "tensorflow"],
+                }
+                if re.match("(?=^ml_)", target_instance_type):
+                    return framework in multi_version_frameworks_support_mapping["ml_ioc"]
```

Comment:
what are the options for instance type. do they have ml_ioc.

Reply:
ioc target means: ml_c4, ml_c5, ml_m4, ml_m5, ml_p2, ml_p3, ml_g4dn, they support torch and tf framework_versions.
ml_inf1 supports torch, tf and mxnet.

I can make it more explicit if you think it is better. (I was trying to make the change minimal)

Reply:
do you want to select ioc first and then inf